osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -285,7 +288,8 @@ public List<BeatmapSetInfo> GetAllUsableBeatmapSets()
         /// <param name="beatmapInfo">The <see cref="BeatmapInfo"/> to save the content against. The file referenced by <see cref="BeatmapInfo.Path"/> will be replaced.</param>
         /// <param name="beatmapContent">The <see cref="IBeatmap"/> content to write.</param>
         /// <param name="beatmapSkin">The beatmap <see cref="ISkin"/> content to write, null if to be omitted.</param>
-        public virtual void Save(BeatmapInfo beatmapInfo, IBeatmap beatmapContent, ISkin? beatmapSkin = null)
+        /// <param name="transferCollections">Whether to transfer the MD5 hashes in collections referencing this beatmap.</param>
```

Comment:
I really really *really* do not like this param. Not only is it a boolean param (which I have a tendency to pick on), but it is also affecting really minuscule details of the save operation, This to me is an indication that this method is overloaded too much (i.e. too many distinct use cases are using it), and therefore this needs to be refactored properly. Tacking boolean flags onto this will just make the situation worse in the long run IMO.

Reply:
Alternatively the `TransferCollectionReferences()` should be removed from `Save`? Maybe that's better called where required rather than being silently performed by `Save()`.

Reply:
No, I don't think that's good either. This places the onus of ensuring properly handling collection on callers which is also not ideal.

The thing is that `Save()` sometimes means "update this beatmap" and sometimes it means "create a brand new one". They should be two separate methods. They may share 99% of their body via split private methods, but they should not